Example/CustomTemplate.js

```diff
@@ -0,0 +1,287 @@
+            setModalState(prevState => ({ ...prevState, isTemplateVisible: false }));
+        });
+
+        CleverTap.addListener(CleverTap.CleverTapCustomFunctionPresent, templateName => {
+            presentInAppModal(templateName, true);
+        });
+
+        return () => {
+            CleverTap.removeListener(CleverTap.CleverTapCustomTemplatePresent);
+            CleverTap.removeListener(CleverTap.CleverTapCustomTemplateClose);
+            CleverTap.removeListener(CleverTap.CleverTapCustomFunctionPresent);
+        };
+    }, []);
+
+    const presentInAppModal = (name, isFunction) => {
+        getTemplateValuesString(name).then((str) => {
+            let description = `Arguments for "${name}":${str}`;
+            setModalState(prevState => {
+                // If the in-app template modal is already visible, this means `presentInAppModal`
+                // was called by a isVisual:false function triggered by an action.
+                if (prevState.isTemplateVisible) {
+                    console.log(`Showing a custom function "${name}" with isVisual:false,
+triggered by action from template "${prevState.templateName}".`);
+                    // Prepare to show the non-visual function popup
+                    return {
+                        ...prevState,
+                        isTemplateVisible: false,
+                        isNonVisualFunctionVisible: true,
+                        functionDescription: description,
+                        functionName: name,
+                    };
+                } else {
+                    // Show the InApp Popup
+                    return {
+                        ...prevState,
+                        isTemplateVisible: true,
+                        isNonVisualFunctionVisible: false,
+                        templateDescription: description,
+                        templateName: name,
+                        isStandaloneFunction: isFunction
+                    };
+                }
+            });
+        });
+    };
+
+    const getTemplateValuesString = async (templateName) => {
+        switch (templateName) {
+            case "Example template":
+                try {
+                    var bool = await CleverTap.customTemplateGetBooleanArg(templateName, "bool");
+                    var string = await CleverTap.customTemplateGetStringArg(templateName, "string");
+                    var mapInt = await CleverTap.customTemplateGetNumberArg(templateName, "map.int");
+                    var mapString = await CleverTap.customTemplateGetStringArg(templateName, "map.string");
+                    var map = await CleverTap.customTemplateGetObjectArg(templateName, "map");
+                    var file = await CleverTap.customTemplateGetFileArg(templateName, "file");
+                    return `
+bool=${bool}
+string=${string}
+map.int=${mapInt}
+map.string=${mapString}
+map=${JSON.stringify(map)}
+file=${file}`;
```

Comment:
Maybe also add the ```action``` argument as a string just to know it is there, but there is currently no way to retrieve the value of it. Maybe we can add a customTemplateGetCurrentContextString (not a very pretty name) to be able to provide templateContext.toString and get all those values as a string ? It would also be easier when the template is modified

Reply:
Added `customTemplateContextToString` to get the `TemplateContext.toString` on Android and `TemplateContext.debugDescription` on iOS.